### PR TITLE
[zh-cn]: Update translation of Object.preventExtensions()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

None.

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md [hunk 1]
```diff
@@ -26,55 +26,61 @@ Object.preventExtensions(obj)
...
 // throws a 'TypeError'.
-fixed.__proto__ = { oh: 'hai' };
+fixed.__proto__ = { oh: "hai" };
 ```
 
-## Notes
+## 非对象参数
 
-在 ES5 中，如果参数不是一个对象类型（而是原始类型），将抛出一个{{jsxref("TypeError")}}异常。在 ES2015 中，非对象参数将被视为一个不可扩展的普通对象，因此会被直接返回。
+在 ES5 中，如果参数不是一个对象类型（而是原始类型），将抛出一个 {{jsxref("TypeError")}} 异常。在 ES2015 中，非对象参数将被视为一个不可扩展的普通对象，因此会被直接返回。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md:83]:
```suggestion
在 ES5 中，如果参数不是一个对象类型（而是原始类型），将抛出 {{jsxref("TypeError")}} 。在 ES2015 中，非对象参数将被视为一个不可扩展的普通对象，因此会被直接返回。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md [hunk 2]
```diff
@@ -26,55 +26,61 @@ Object.preventExtensions(obj)
...
 // throws a 'TypeError'.
-fixed.__proto__ = { oh: 'hai' };
+fixed.__proto__ = { oh: "hai" };
 ```
 
-## Notes
+## 非对象参数
 
-在 ES5 中，如果参数不是一个对象类型（而是原始类型），将抛出一个{{jsxref("TypeError")}}异常。在 ES2015 中，非对象参数将被视为一个不可扩展的普通对象，因此会被直接返回。
+在 ES5 中，如果参数不是一个对象类型（而是原始类型），将抛出 {{jsxref("TypeError")}} 。在 ES2015 中，非对象参数将被视为一个不可扩展的普通对象，因此会被直接返回。
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md:83]:
```suggestion
在 ES5 中，如果参数不是一个对象类型（而是原始类型），将抛出 {{jsxref("TypeError")}}。在 ES2015 中，非对象参数将被视为一个不可扩展的普通对象，因此会被直接返回。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md [hunk 3]
```diff
@@ -26,55 +26,61 @@ Object.preventExtensions(obj)
...
+  value: 8675309,
+}); // 抛出 TypeError
 
 // 在严格模式中，为一个不可扩展对象的新属性赋值会抛出 TypeError 异常。
-function fail()
-{
+function fail() {
   "use strict";
-  nonExtensible.newProperty = "FAIL"; // throws a TypeError
+  // throws a TypeError
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md:67]:
```suggestion
  // 抛出 TypeError
```

File: files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md [hunk 4]
```diff
@@ -26,55 +26,61 @@ Object.preventExtensions(obj)
 
 ## 描述
 
-如果一个对象可以添加新的属性，则这个对象是可扩展的。`Object.preventExtensions()` 将对象标记为不再可扩展，这样它将永远不会具有它被标记为不可扩展时持有的属性之外的属性。注意，一般来说，不可扩展对象的属性可能仍然可被*删除*。尝试将新属性添加到不可扩展对象将静默失败或抛出 {{jsxref("TypeError")}}（最常见的情况是在{{jsxref("Functions_and_function_scope/Strict_mode", "严格模式", "", 1)}}中，但不排除其他情况）。
+如果一个对象可以添加新的属性，则这个对象是可扩展的。`Object.preventExtensions()` 将对象标记为不再可扩展，这样它将永远不会具有它被标记为不可扩展时持有的属性之外的属性。注意，一般来说，不可扩展对象的属性仍然可以被*删除*。尝试向不可扩展对象添加新属性将静默失败或抛出 {{jsxref("TypeError")}}（最常见的情况是在{{jsxref("Functions_and_function_scope/Strict_mode", "严格模式", "", 1)}}中，但不排除其他情况）。
 
-`Object.preventExtensions()` 仅阻止添加自身的属性。但其对象类型的原型依然可以添加新的属性。
+与 [`Object.seal()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/seal) 和 [`Object.freeze()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) 不同，`Object.preventExtensions()` 调用了内在的 JavaScript 行为，不能用几个其他操作的组合替代。它还有它的 `Reflect` 对应方法（仅存在于内在操作中），[`Reflect.preventExtensions()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/preventExtensions)。
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md:31]:
```suggestion
与 [`Object.seal()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/seal) 和 [`Object.freeze()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) 不同，`Object.preventExtensions()` 调用了内在的 JavaScript 行为，不能用几个其他操作的组合替代。它还有它的 `Reflect` 对应方法（仅存在于内在操作中），[`Reflect.preventExtensions()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Reflect/preventExtensions)。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md [hunk 5]
```diff
@@ -26,55 +26,61 @@ Object.preventExtensions(obj)
...
 
-如果一个对象可以添加新的属性，则这个对象是可扩展的。`Object.preventExtensions()` 将对象标记为不再可扩展，这样它将永远不会具有它被标记为不可扩展时持有的属性之外的属性。注意，一般来说，不可扩展对象的属性可能仍然可被*删除*。尝试将新属性添加到不可扩展对象将静默失败或抛出 {{jsxref("TypeError")}}（最常见的情况是在{{jsxref("Functions_and_function_scope/Strict_mode", "严格模式", "", 1)}}中，但不排除其他情况）。
+如果一个对象可以添加新的属性，则这个对象是可扩展的。`Object.preventExtensions()` 将对象标记为不再可扩展，这样它将永远不会具有它被标记为不可扩展时持有的属性之外的属性。注意，一般来说，不可扩展对象的属性仍然可以被*删除*。尝试向不可扩展对象添加新属性将静默失败或抛出 {{jsxref("TypeError")}}（最常见的情况是在{{jsxref("Functions_and_function_scope/Strict_mode", "严格模式", "", 1)}}中，但不排除其他情况）。
 
-`Object.preventExtensions()` 仅阻止添加自身的属性。但其对象类型的原型依然可以添加新的属性。
+与 [`Object.seal()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/seal) 和 [`Object.freeze()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) 不同，`Object.preventExtensions()` 调用了内在的 JavaScript 行为，不能用几个其他操作的组合替代。它还有它的 `Reflect` 对应方法（仅存在于内在操作中），[`Reflect.preventExtensions()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Reflect/preventExtensions)。
+
+`Object.preventExtensions()` 只能防止添加自身的属性。但其对象类型的原型依然可以添加新的属性。
 
 该方法使得目标对象的 `[[prototype]]` 不可变；任何重新赋值 `[[prototype]]` 操作都会抛出 `TypeError` 。这种行为只针对内部的 `[[prototype]]` 属性，目标对象的其他属性将保持可变。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md:35]:
```suggestion
该方法使得目标对象的 `[[Prototype]]` 不可变；任何重新赋值 `[[Prototype]]` 操作都会抛出 `TypeError` 。这种行为只针对内部的 `[[Prototype]]` 属性，目标对象的其他属性将保持可变。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md [hunk 6]
```diff
@@ -26,55 +26,61 @@ Object.preventExtensions(obj)
...
+const obj2 = Object.preventExtensions(obj);
+obj === obj2; // true
 
 // 字面量方式定义的对象默认是可扩展的。
-var empty = {};
-Object.isExtensible(empty) //=== true
+const empty = {};
+Object.isExtensible(empty); // true
 
 // ...但可以改变。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md:53]:
```suggestion
// 可以将其改变为不可扩展的。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md [hunk 7]
```diff
@@ -26,55 +26,61 @@ Object.preventExtensions(obj)
...
 // throws a 'TypeError'.
-fixed.__proto__ = { oh: 'hai' };
+fixed.__proto__ = { oh: "hai" };
 ```
 
-## Notes
+## 非对象参数
 
-在 ES5 中，如果参数不是一个对象类型（而是原始类型），将抛出一个{{jsxref("TypeError")}}异常。在 ES2015 中，非对象参数将被视为一个不可扩展的普通对象，因此会被直接返回。
+在 ES5 中，如果参数不是一个对象类型（而是原始类型），将抛出 {{jsxref("TypeError")}}。在 ES2015 中，非对象参数将被视为一个不可扩展的普通对象，因此会被直接返回。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md:83]:
```suggestion
在 ES5 中，如果提供给该方法的参数不是一个对象类型（而是原始类型），将抛出 {{jsxref("TypeError")}}。在 ES2015 中，非对象参数将被原样返回，因为原始类型是定义上不可变的。
```

**Review by jasonren0403 — COMMENTED** [review]
some suggestions

File: files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md [hunk 8]
```diff
@@ -26,55 +26,61 @@ Object.preventExtensions(obj)
...
 ## 描述
 
-如果一个对象可以添加新的属性，则这个对象是可扩展的。`Object.preventExtensions()` 将对象标记为不再可扩展，这样它将永远不会具有它被标记为不可扩展时持有的属性之外的属性。注意，一般来说，不可扩展对象的属性可能仍然可被*删除*。尝试将新属性添加到不可扩展对象将静默失败或抛出 {{jsxref("TypeError")}}（最常见的情况是在{{jsxref("Functions_and_function_scope/Strict_mode", "严格模式", "", 1)}}中，但不排除其他情况）。
+如果一个对象可以添加新的属性，则这个对象是可扩展的。`Object.preventExtensions()` 将对象标记为不再可扩展，这样它将永远不会具有它被标记为不可扩展时持有的属性之外的属性。注意，一般来说，不可扩展对象的属性仍然可以被*删除*。尝试向不可扩展对象添加新属性将静默失败，或在[严格模式](/zh-CN/docs/Web/JavaScript/Reference/Strict_mode)中抛出 {{jsxref("TypeError")}}。
 
-`Object.preventExtensions()` 仅阻止添加自身的属性。但其对象类型的原型依然可以添加新的属性。
+与 [`Object.seal()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/seal) 和 [`Object.freeze()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) 不同，`Object.preventExtensions()` 调用了内在的 JavaScript 行为，不能用几个其他操作的组合替代。它还有它的 `Reflect` 对应方法（仅存在于内在操作中），[`Reflect.preventExtensions()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Reflect/preventExtensions)。
 
-该方法使得目标对象的 `[[prototype]]` 不可变；任何重新赋值 `[[prototype]]` 操作都会抛出 `TypeError` 。这种行为只针对内部的 `[[prototype]]` 属性，目标对象的其他属性将保持可变。
+`Object.preventExtensions()` 只能防止添加自身的属性。但其对象类型的原型依然可以添加新的属性。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/preventextensions/index.md:33]:
```suggestion
`Object.preventExtensions()` 只能防止添加自有属性。但其对象类型的原型依然可以添加新的属性。
```

**Review by yin1999 — APPROVED** [review]
Thanks, just one nit ;)

**Pull request merged**